Reverts e6ac87d7a291 (common.mak: Remove NO_PIE_LDFLAGS) since with the
alpine toolchain -static does not imply -no-pie

--- a/rust/pvimg/boot/Makefile
+++ b/rust/pvimg/boot/Makefile
@@ -74,7 +74,7 @@ stage3b.elf: head.o $(ZIPL_OBJS)
 
 .SECONDARY: $(FILES:.bin=.lds)
 %.elf: %.lds %.o
-	$(LINK) $(NO_WARN_RWX_SEGMENTS_LDFLAGS) -Wl,-T,$< -Wl,--build-id=none -m64 -static -nostdlib $(filter %.o, $^) -o $@
+	$(LINK) -no-pie $(NO_WARN_RWX_SEGMENTS_LDFLAGS) -Wl,-T,$< -Wl,--build-id=none -m64 -static -nostdlib $(filter %.o, $^) -o $@
 	@chmod a-x $@
 
 %.bin.debug: %.elf
--- a/zipl/boot/Makefile
+++ b/zipl/boot/Makefile
@@ -93,7 +93,7 @@ stage3.elf: head.o stage3.o kdump3.o libc.o ebcdic.o ebcdic_conv.o sclp.o \
 	     sclp_stage3.o kdump.o entry.o stage3.lds
 
 %.elf:	%.o
-	$(LINK) -Wl,-T,$(filter %.lds,$^) $(NO_WARN_RWX_SEGMENTS_LDFLAGS) -Wl,--build-id=none -m64 -static -nostdlib $(filter %.o, $^) -o $@
+	$(LINK) -Wl,-T,$(filter %.lds,$^) -no-pie $(NO_WARN_RWX_SEGMENTS_LDFLAGS) -Wl,--build-id=none -m64 -static -nostdlib $(filter %.o, $^) -o $@
 	@chmod a-x $@
 
 %.bin.debug.elf: %.elf
